chain/jsonrpc/jsonrpc-tests/tests/rpc_transactions.rs

```diff
@@ -97,6 +98,34 @@ fn test_send_tx_commit() {
     });
 }
 
+/// Test get_recursive_transaction_results (called by get_final_transaction_result)
+/// only returns non-refund receipts
+#[test]
+fn test_refunds_not_in_receipts() {
+    test_with_client!(test_utils::NodeType::Validator, client, async move {
+        let block_hash = client.block(BlockReference::latest()).await.unwrap().header.hash;
+        let signer = InMemorySigner::from_seed("test1".parse().unwrap(), KeyType::ED25519, "test1");
+        let tx = SignedTransaction::send_money(
+            1,
+            "test1".parse().unwrap(),
+            "test2".parse().unwrap(),
+            &signer,
+            100,
+            block_hash,
+        );
+        let bytes = tx.try_to_vec().unwrap();
+        client.broadcast_tx_commit(to_base64(&bytes)).await.unwrap();
+        thread::sleep(time::Duration::from_secs(5));
+        let tx_status = client.EXPERIMENTAL_tx_status(to_base64(&bytes)).await.unwrap();
+        for receipt in tx_status.get("receipts") {
+            if !receipt.as_array().unwrap().is_empty() {
+                let is_not_refund = !receipt.get("predecessor_id").is_none();
+                assert!(is_not_refund);
+            }
+        }
```

Comment:
Hm, does this actually test the desired behavior? I seem to always get an empty array of recipts, both with and without the change?

Reply:
Forgot the `is_system`. Added in updated commit.